src/drivers/storage/SDCard.cpp

```diff
@@ -167,7 +169,12 @@ bool SDCard::initSDcard()
 #elif defined (BUILD_SDMMC_1)
 #warning SDMMC : 1 - bit mode is not always working. If you experience issues, try other modes.
         iSD_->setPins(SDMMC_CLK, SDMMC_CMD, SDMMC_D0);
+#ifdef NERDMINER_T_HMI
```

Comment:
you could define a SD_FREQUENCY variable
so it could be tunable for every board

Reply:
A '#define'  seems reasonable here
```
#define SDMMC_FREQ_20MHZ  20000
#define SDMMC_FREQ_10MHZ  10000
#define SDMMC_FREQ_5MHZ   5000
#define SDMMC_FREQ_2MHZ   2000
```
but a class refactor of some sort would be needed if I understand the suggestion.
```
class SDCard {
public:
    SDCard(uint32_t frequency = SDMMC_FREQ_20MHZ);
    void setFrequency(uint32_t frequency);

private:
    uint32_t frequency;
};
```
https://github.com/espressif/esp-idf/blob/0453e8608bde98133a427a74ae61d272770b1bfd/components/sdmmc/include/sd_protocol_types.h#L187

Reply:
> A '#define' seems reasonable here
> 
> ```
> #define SDMMC_FREQ_20MHZ  20000
> #define SDMMC_FREQ_10MHZ  10000
> #define SDMMC_FREQ_5MHZ   5000
> #define SDMMC_FREQ_2MHZ   2000
> ```
> 
> but a class refactor of some sort would be needed if I understand the suggestion.
> 
> ```
> class SDCard {
> public:
>     SDCard(uint32_t frequency = SDMMC_FREQ_20MHZ);
>     void setFrequency(uint32_t frequency);
> 
> private:
>     uint32_t frequency;
> };
> ```
> 
> https://github.com/espressif/esp-idf/blob/0453e8608bde98133a427a74ae61d272770b1bfd/components/sdmmc/include/sd_protocol_types.h#L187

Dear,

#ifdef SD_FREQUENCY
        cardInitialized_ = iSD_->begin("/sd", true, false, SD_FREQUENCY);
#else
        cardInitialized_ = iSD_->begin("/sd", true);
#endif

and you can add -D SD_FREQUENCY=20000 inside platformio.ini or #define SD_FREQUENCY 20000 inside lilygoT_HMI.h

Reply:
simplicity is elegant